HermiT issues

After we discussed the imports issue for HermiT in https://github.com/ease-crc/ease_ontology/pull/40#issuecomment-586211855 and once the fix is applied in #56, there are still other issues HermiT brings up.

The new Python script produces following output (I used HermiT 1.3.8 from http://www.hermit-reasoner.com/ if I see it correctly):
```
Exception in thread "main" org.semanticweb.HermiT.datatypes.UnsupportedDatatypeException: HermiT supports all and only the datatypes of the OWL 2 datatype map, see 
http://www.w3.org/TR/owl2-syntax/#Datatype_Maps. 
The datatype 'http://www.ease-crc.org/ont/EASE.owl#array_double' is not part of the OWL 2 datatype map and 
no custom datatype definition is given; 
therefore, HermiT cannot handle this datatype.
	at org.semanticweb.HermiT.datatypes.DatatypeRegistry.getDatatypeHandlerFor(Unknown Source)
	at org.semanticweb.HermiT.datatypes.DatatypeRegistry.getDatatypeHandlerFor(Unknown Source)
	at org.semanticweb.HermiT.datatypes.DatatypeRegistry.validateDatatypeRestriction(Unknown Source)
	at org.semanticweb.HermiT.structural.OWLClausification$DataRangeConverter.visit(Unknown Source)
	at uk.ac.manchester.cs.owl.owlapi.OWLDatatypeImpl.accept(OWLDatatypeImpl.java:343)
	at org.semanticweb.HermiT.structural.OWLClausification$DataRangeConverter.convertDataRange(Unknown Source)
	at org.semanticweb.HermiT.structural.OWLClausification$NormalizedAxiomClausifier.visit(Unknown Source)
	at uk.ac.manchester.cs.owl.owlapi.OWLDataMaxCardinalityImpl.accept(OWLDataMaxCardinalityImpl.java:88)
	at org.semanticweb.HermiT.structural.OWLClausification.clausify(Unknown Source)
	at org.semanticweb.HermiT.structural.OWLClausification.preprocessAndClausify(Unknown Source)
	at org.semanticweb.HermiT.Reasoner.loadOntology(Unknown Source)
	at org.semanticweb.HermiT.Reasoner.<init>(Unknown Source)
	at org.semanticweb.HermiT.Reasoner.<init>(Unknown Source)
	at org.semanticweb.HermiT.cli.CommandLine.main(Unknown Source)
```

If running from inside Protégé, HermiT succeeds (except the consistency issues mentioned in #40).

If I use the version shipped with Protégé (e.g. for me `python3 scripts/hermit_test.py -r /Applications/Protégé.app/Contents/Java/plugins/org.semanticweb.hermit-1.4.3.456.jar -o output owl/EASE-UGLY.owl`), the jar file cannot be used as it is missing the main manifest attribute:
```
ERROR: reasoner failed to complete classification query
no main manifest attribute, in /Applications/Prot\xc3\xa9g\xc3\xa9.app/Contents/Java/plugins/org.semanticweb.hermit-1.4.3.456.jar
```

Which version do you use to run the python script?

On my fork I am now experimenting with building HermiT from source to run it properly (https://github.com/shoeffner/ease_ontology/tree/hermit_ci -- both versions are in that branch).
With version 1.4.4.519 from https://github.com/owlcs/hermit-reasoner/commit/57b79367f2b66f0b510598474efbde43195410cc, I get the same error as above (array_double is not part of OWL data type map, https://github.com/shoeffner/ease_ontology/runs/450978925), with the unknown version (probably 1.3.8, it's 1.4.0-SNAPSHOT) from https://github.com/phillord/hermit-reasoner/commit/37ec30aced32ac81ebecc5e33fad255ddefcb4c3, I get the following issue (https://github.com/shoeffner/ease_ontology/runs/451281407):

```
Exception in thread "main" java.lang.NoClassDefFoundError: rationals/NoSuchStateException
	at org.semanticweb.HermiT.structural.OWLClausification.preprocessAndClausify(OWLClausification.java:148)
	at org.semanticweb.HermiT.Reasoner.loadOntology(Reasoner.java:221)
	at org.semanticweb.HermiT.Reasoner.<init>(Reasoner.java:212)
	at org.semanticweb.HermiT.Reasoner.<init>(Reasoner.java:187)
	at org.semanticweb.HermiT.cli.CommandLine.main(CommandLine.java:761)
Caused by: java.lang.ClassNotFoundException: rationals.NoSuchStateException
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 5 more
```

Note that I applied patches to the pom files for both versions: For both I added to steps necessary to build a jar with dependencies and for the upstream repo (phillord) I removed the svn versioning plugin and publishing plugins.

So I can't get HermiT to run except from inside Protégé, do you have any pointers which version works for you?

--- a/owl/EASE.owl
+++ b/owl/EASE.owl
@@ -81,37 +81,49 @@ A style recommendation for usage guidelines is that they should be brief. In con
 
     <!-- http://www.ease-crc.org/ont/EASE.owl#array_boolean -->
 
-    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_boolean"/>
+    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_boolean">
+        <owl:equivalentClass rdf:resource="http://www.w3.org/2001/XMLSchema#string"/>
+    </rdfs:Datatype>
     
 
 
     <!-- http://www.ease-crc.org/ont/EASE.owl#array_double -->
 
-    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_double"/>
+    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_double">
+        <owl:equivalentClass rdf:resource="http://www.w3.org/2001/XMLSchema#string"/>
+    </rdfs:Datatype>
     
 
 
     <!-- http://www.ease-crc.org/ont/EASE.owl#array_float -->
 
-    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_float"/>
+    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_float">
+        <owl:equivalentClass rdf:resource="http://www.w3.org/2001/XMLSchema#string"/>
+    </rdfs:Datatype>
     
 
 
     <!-- http://www.ease-crc.org/ont/EASE.owl#array_int -->
 
-    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_int"/>
+    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_int">
+        <owl:equivalentClass rdf:resource="http://www.w3.org/2001/XMLSchema#string"/>
+    </rdfs:Datatype>
     
 
 
     <!-- http://www.ease-crc.org/ont/EASE.owl#array_string -->
 
-    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_string"/>
+    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_string">
+        <owl:equivalentClass rdf:resource="http://www.w3.org/2001/XMLSchema#string"/>
+    </rdfs:Datatype>
     
 
 
     <!-- http://www.ease-crc.org/ont/EASE.owl#array_uint -->
 
-    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_uint"/>
+    <rdfs:Datatype rdf:about="http://www.ease-crc.org/ont/EASE.owl#array_uint">
+        <owl:equivalentClass rdf:resource="http://www.w3.org/2001/XMLSchema#string"/>
+    </rdfs:Datatype>
     
 
 

